lib/suspenders/adapters/heroku.rb

```diff
@@ -0,0 +1,52 @@
+module Suspenders
+  module Adapters
+    class Heroku
+      def initialize(app_builder)
+        @app_builder = app_builder
```

Comment:
I did not know about the warning. I believe the reasoning behind them being protected is for subclassing, but I never found myself in the need to make a reader protected yet (and always defaulted to private).

The reason behind using getters instead of instance variables is that they make refactoring, renaming and extending objects easier.